### PR TITLE
void

### PR DESCRIPTION
**Pull request by abebeos**

### CODE REVIEW

**Comment by algorandskiy** [comment]
what is a point of this PR?

**Comment by abebeos** [comment]
> what is a point of this PR?

Quick answer: code quality.

Packaging related code into a package  (via strict refactoring, zero functional changes) is nearly always a valid  enhancement.

I did it here because in order to fulfil #5088, the `formatting` related code needed to be packaged first.

**Comment by abebeos** [comment]
I've updated the original test ("Rationale" section) so the PR becomes better documented.

**Comment by abebeos** [comment]
> what is a point of this PR?

The update PR should now "proof its point"...

**Comment by abebeos** [comment]
> what is a point of this PR?

@algorandskiy  The update PR should now "proof its point"...

File: cmd/goal/messages.go [hunk 1]
```diff
@@ -24,8 +24,8 @@ const (
 	errorGenesisIDFail          = "Error determining kmd folder (%s). Ensure the node is running in %s."
 	errorDirectoryNotExist      = "Specified directory '%s' does not exist."
 	errorParseAddr              = "Failed to parse addr: %v"
-	errorNonPrintableCharacters = "One or more non-printable characters were omitted from the following error message:"
-	infoNonPrintableCharacters  = "One or more non-printable characters were omitted from the subsequent line:"
+//	errorNonPrintableCharacters = "One or more non-printable characters were omitted from the following error message:"
```
Comment (bbroder-algo) [inline cmd/goal/messages.go:27]: just remove them?

File: cmd/util/report/report.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
```
Comment (bbroder-algo) [inline cmd/util/report/report.go:1]: mostly former cmd/util/goal/commands.go impls

**Comment by bbroder-algo** [comment]
ok I eyeballed every change to verify that the log category hadn't changed and I found none that had.  Where do I send my optometry bill?

**Comment by bbroder-algo** [comment]
some of the cmd implementations of reportInfoLn differed from each other, such as in cmd/goal/commands.go vs cmd/catchpointdump/commands.go

File: cmd/util/report/report.go (hunk 2 above)
Comment (abebeos) [reply cmd/util/report/report.go:1]: > mostly former cmd/util/goal/commands.go impls

Correct, they had additionally the unicodeprintable check (`printable, line := formatting.UnicodePrintable(line)`), so used them.

**Comment by abebeos** [comment]
> Where do I send my optometry bill?

I guess me? Possibly I squashed the last part too much...

**Comment by abebeos** [comment]
> some of the cmd implementations of reportInfoLn differed from each other, such as in cmd/goal/commands.go vs cmd/catchpointdump/commands.go

`goal` adds some unicode-printable check that looked of "general use" to me, so picked it in favor if the simple ones (e.g. within `catchpointdump`)

https://github.com/algorand/go-algorand/blob/6487b374ca303e98317739a5f129b12a35f76176/cmd/goal/commands.go#L442-L450

https://github.com/algorand/go-algorand/blob/6487b374ca303e98317739a5f129b12a35f76176/cmd/catchpointdump/commands.go#L105-L107

File: cmd/goal/messages.go (hunk 1 above)
Comment (abebeos) [reply cmd/goal/messages.go:27]: Moved then to the report package

https://github.com/algorand/go-algorand/blob/ea410c8bd25e800498c20272ac2c0950d7adf675/cmd/util/report/messages.go#L17-L22

Comment (abebeos) [reply cmd/goal/messages.go:27]: > just remove them?

oh, it seems you meant "remove them instead of out-comment them".

k.

**Comment by abebeos** [comment]
( @bbroder-algo , maybe you could take a look at #5108, too? - A chirurgical tiny patch, compared to the eye-straining one here)

**Comment by bbroder-algo** [comment]
> `goal/commands.go` version adds some unicode-printable check that looked of "general use" to me, so picked it in favor if the simple ones (e.g. within `catchpointdump`)

I suppose my thought here is that it isn't quite the claimed "zero functional changes".

**Comment by abebeos** [comment]
> > `goal/commands.go` version adds some unicode-printable check that looked of "general use" to me, so picked it in favor if the simple ones (e.g. within `catchpointdump`)
> 
> I suppose my thought here is that it isn't quite the claimed "zero functional changes".

Right. I've added a note to the summary!

**Comment by abebeos** [comment]
> It would also be helpful to update the configuration for our govet linter settings, where we tell it about custom functions that are printf- or println-derived, so it can catch mistakes with mismatched printf format strings and args, or if someone uses printf instead of println (or vice-versa). I think this would do it:

Ok, ty!

I've added a note to the original issue, to process this right after the merge together with some other tiny debt.

File: cmd/goal/commands.go [hunk 3]
```diff
@@ -204,10 +204,10 @@ var reportCmd = &cobra.Command{
 		fmt.Println(string(data))
 
 		dirs := datadir.GetDataDirs()
-		report := len(dirs) > 1
+		hasDirs := len(dirs) > 1
 		for _, dir := range dirs {
-			if report {
-				reportInfof(infoDataDir, dir)
+			if hasDirs {
```
Comment (bbroder-algo) [inline cmd/goal/commands.go:209]: this bit is to prevent the namespace collision

**Comment by bbroder-algo** [comment]
I created a patch file with `git diff -U0 7f7939 > patch_file.diff` and ran this script on it to automatically check the rewrites
```
import difflib
from patch import fromfile
import re

diff_file = "patch_file.diff"
diff_patch = fromfile(diff_file)
passes, fails, errors, others = [], [], [], []
for patch in diff_patch:
    for hunk in patch:
        removeadds, removals = [], []
        for line in hunk.text:
            if line[0] == ord('+'): 
                if len(removals) == 0: continue
                removeadds.append ( (removals, line) )
                removals = []
            else:
                removals.append (line)

        logwords = "(Error|Info|Warn|Debug|Trace)" 
        pattern = r"report" + logwords
        matched = False
        for ra in removeadds:
            for pot in [ (str(a), str(ra[1])) for a in ra[0] if re.search (pattern, str(a)) ]:
                matched = True
                elim, added = pot[0][3:], pot[1][3:]
                for i, c in enumerate(elim):
                    if i >= len(added) or c != added[i]:
                       break
                elim, added = elim[i:], added[i:]
                
                if not(re.search ("^"+logwords, elim)):
                    errors.append ( (elim, added, patch, hunk) )
                elif (elim == added[1:]):
                    passes.append ( (elim, added, patch, hunk) )
                else:
                    fails.append ( (elim, added, patch, hunk) )
        if not matched: others.append ( (patch, hunk) )

print ("Failing:")
for fail in fails:
    print (fail[2].header[0], '@@ -%s,%s +%s,%s @@' % (fail[3].startsrc, fail[3].linessrc, fail[3].starttgt, fail[3].linestgt))
print ("\nErrors:")
for err in errors:
    print (err[2].header[0], '@@ -%s,%s +%s,%s @@' % (err[3].startsrc, err[3].linessrc, err[3].starttgt, err[3].linestgt))
print ("\nPassing: ", len(passes))
print ("\nOther:")
for oth in others:
    print (oth[0].header[0], '@@ -%s,%s +%s,%s @@' % (oth[1].startsrc, oth[1].linessrc, oth[1].starttgt, oth[1].linestgt))
    for l in oth[1].text:
         print ("    " + str (l))
```
It identified 6 different equality failures and errors:
```
Failing:
b'diff --git a/cmd/goal/application.go b/cmd/goal/application.go\n' @@ -497,2 +499,2 @@
b'diff --git a/cmd/goal/application.go b/cmd/goal/application.go\n' @@ -572,2 +574,2 @@
b'diff --git a/cmd/goal/asset.go b/cmd/goal/asset.go\n' @@ -801,3 +802,3 @@
b'diff --git a/cmd/goal/asset.go b/cmd/goal/asset.go\n' @@ -801,3 +802,3 @@
b'diff --git a/cmd/goal/box.go b/cmd/goal/box.go\n' @@ -86,1 +88,1 @@
b'diff --git a/cmd/goal/wallet.go b/cmd/goal/wallet.go\n' @@ -169,2 +170,2 @@

Errors:
b'diff --git a/cmd/goal/box.go b/cmd/goal/box.go\n' @@ -114,2 +116,2 @@
b'diff --git a/cmd/goal/commands.go b/cmd/goal/commands.go\n' @@ -209,2 +209,2 @@
```
When those were hand-checked, they revealed the script can't handle accurate changes to line-sequential log calls, or other acceptable issues (using the new formatting package, etc).

I hand checked all the other non-matching changes in this PR and they passed as proper moves or refactorings.  I also re-re-checked util/report/report.go against goal/commands.go.  

I think libgoal should also be modified to use util/datadir.

I can't find anything wrong with this PR.  Perhaps a remaining question is whether it is all worth it.

**Comment by abebeos** [comment]
> I can't find anything wrong with this PR. 

seeing your review effort (ty, as a sidenote...), maybe I should have send in more commits. I can still "replay" the last commit, separating e.g. the search&replace part into dedicated commits (though this would mean in essence that review starts from scratch).

> Perhaps a remaining question is whether it is all worth it.

maintenance, hidden bugs, CLI policy enforcement etc. etc. - copy&paste is not the way to go.

**Review by bbroder-algo — APPROVED** [review]
I challenge anyone to find what's wrong here.

**Comment by abebeos** [comment]
> I challenge anyone to find what's wrong here.

I have provide the simpler to review version in a separate PR:

https://github.com/algorand/go-algorand/pull/5125/commits

That PR sticks to the most important rule: to isolate the search&replace-changes in one dedicated commit. Not exactly sure what happened that I violated that rule in #5090 here.

(I am too afraid to push it here, to not mess up the already done review)

**Comment by abebeos** [comment]
> I think libgoal should also be modified to use util/datadir.

I limited the scope of this PR to `cmd/goal`. Once merged, next step would be to polish the new packages, and then consume them broader within `cmd/...`. Final step to introduce them in the overall codebase.

(never again with a squashed commit though...)

**Comment by abebeos** [comment]
@bbroder-algo , I dislike myself those mega-PRs, so I understand if it takes possibly longer to get 2 more reviews. I provided the unsquashed #5125 for easier review.

(my problem here is that I'm stuck with the follow-up tasks)

**Comment by abebeos** [comment]
@winder , @jasonpaulos , I've unsqaushed the big commit and reapplied it in 4 commits to simplify review. Should be now easy to review, due to the refactoring nature of the PR.

**Comment by abebeos** [comment]
@Eric-Warehime  @algorandskiy , https://github.com/algorand/go-algorand/pull/5090#pullrequestreview-1289811368

**Comment by abebeos** [comment]
@bbroder-algo , I withdraw this one and ask that it will not be merged.

It is not acceptable that some clowns (from the Algorand Foundation) have the freedom to exploit bounty-workers, just to tell them at some point that their work was just worth 500 ALGO, and that is was anyways out of scope ("bug/feature").

**Comment by bbroder-algo** [comment]
ok no problem thanks for your work here hopefully conditions will be better in the future for your contributions.

**Comment by abebeos** [comment]
@bbroder-algo , I thank you for your quick and thorough reviews. I noticed that you're kind of a project-lead here. I assure you that the Foundation alienates people. I have no problems to do some work for free now and then, but not when bounties are advertised.

Whenever you need a break from code, please take a look at this:

https://github.com/systemic-flaws/dapp-platforms

You really really need to fix the Algorand Foundation. Sticking to standard-procedures is all that's needed.

(I've still not recovered mentally from this "Please note our financial team processes payments bi-monthly", I'm close to hitting my head against the wall...)

**Pull request closed without merging**